src/adapter/src/client.rs

```diff
@@ -254,6 +262,57 @@ Issue a SQL query to get started. Need help?
             )));
         }
 
+        // Users stub their toe on their default cluster not existing, so we provide a notice to
+        // help guide them on what do to.
+        let cluster_var = session
+            .vars()
+            .inspect(CLUSTER.name())
+            .expect("cluster should exist");
+        if catalog.resolve_cluster(Some(&cluster_active)).is_err() {
+            let cluster_notice = 'notice: {
+                // If the user provided a cluster via a connection configuration parameter, do not
```

Comment:
What's the motivation behind this? Could the intention also be described in the comment here, because it's not obvious?

Reply:
Good call, updated with the reasoning!